docs/11_search.md

```diff
@@ -0,0 +1,286 @@
+
+### Whitelisting search attributes
+
+Start by querying the list of search attributes using the CLI:
+
+```bash
+$ cadence --domain samples-domain cl get-search-attr
++---------------------+------------+
+|         KEY         | VALUE TYPE |
++---------------------+------------+
+| CloseStatus         | INT        |
+| CloseTime           | INT        |
+| CustomBoolField     | DOUBLE     |
+| CustomDatetimeField | DATETIME   |
+| CustomDomain        | KEYWORD    |
+| CustomDoubleField   | BOOL       |
+| CustomIntField      | INT        |
+| CustomKeywordField  | KEYWORD    |
+| CustomStringField   | STRING     |
+| DomainID            | KEYWORD    |
+| ExecutionTime       | INT        |
+| HistoryLength       | INT        |
+| RunID               | KEYWORD    |
+| StartTime           | INT        |
+| WorkflowID          | KEYWORD    |
+| WorkflowType        | KEYWORD    |
++---------------------+------------+
+```
+
+Note that the `--domain` option doesn't affect the output of the results as search attributes are global to the Cadence cluster. There is an [open issue](https://github.com/uber/cadence/issues/2329) to fix this.
+
+Use the admin CLI to add a new search attribute:
+
+```bash
+cadence --domain samples-domain adm cl asa --search_attr_key NewKey --search_attr_type 1
+```
+
+The numbers for the attribute types map as follows:
+
+- 0 = String
+- 1 = Keyword
+- 2 = Int
+- 3 = Double
+- 4 = Bool
+- 5 = DateTime
+
+#### Keyword vs String
+
+Note that **Keyword** and **String** are concepts taken from ElasticSearch. Each word in a **String** is considered a searchable keyword. For a UUID, that can be problematic as ElasticSearch will index each portion of the UUID separately. To have the whole string considered as a searchable keyword, use the **Keyword** type.
+
+For example, key RunID with value "2dd29ab7-2dd8-4668-83e0-89cae261cfb1"
+
+- as a **Keyword** will only be matched by RunID = "2dd29ab7-2dd8-4668-83e0-89cae261cfb1" (or in the future with [regular expressions](https://github.com/uber/cadence/issues/1137))
+- as a **String** will be matched by RunID =  "2dd8", which may cause unwanted matches
+
+**Note:** String type can not be used in Order By query.
+
+There are some pre-whitelisted search attributes that are handy for testing:
+
+- CustomKeywordField
+- CustomIntField
+- CustomDoubleField
+- CustomBoolField
+- CustomDatetimeField
+- CustomStringField
+
+Their types are indicated in their names.
+
+### Value types
+
+Here are the Search Attribute value types and their correspondent Golang types:
+
+- Keyword = string
+- Int = int64
+- Double = float64
+- Bool = bool
+- Datetime = time.Time
+- String = string
+
+### Limit
+
+Cadence recommends limiting the number of ElasticSearch indexes by enforcing limits on the following:
```

Comment:
Unless Cadence *actually* somehow makes this recommendation, it should be worded as "We recommend"